docs/plugin-development.rst

```diff
@@ -129,42 +131,79 @@ will default to being restricted to administrators.
 When you start the server, you may notice a warning message appears:
 ``WARNING: No description docs present for route GET item/:id/cat``. You
 can add self-describing API documentation to your route using the
-``describeRoute`` decorator and ``Description`` class as in the following
-example: ::
+``autoDescribeRoute`` decorator and :py:class:`girder.api.describe.Description` class as in the following
+example:
+
+.. code-block:: python
 
-    from girder.api.describe import Description, describeRoute
+    from girder.api.describe import Description, autoDescribeRoute
     from girder.api import access
 
     @access.public
-    @describeRoute(
+    @autoDescribeRoute(
         Description('Retrieve the cat for a given item.')
         .param('id', 'The item ID', paramType='path')
         .param('cat', 'The cat value.', required=False)
         .errorResponse())
-    def myHandler(id, params):
+    def myHandler(id, cat, params):
         return {
            'itemId': id,
-           'cat': params.get('cat', 'No cat param passed')
+           'cat': cat
         }
 
-
 That will make your route automatically appear in the Swagger documentation
 and will allow users to interact with it via that UI. See the
 :ref:`RESTful API docs<restapi>` for more information about the Swagger page.
+In addition, the ``autoDescribeRoute`` decorator handles a lot of the validation
+and type coercion for you, with the benefit of ensuring that the documentation of
+the endpoint inputs matches their actual behavior. Documented parameters will be
+sent to the method as kwargs (so the order you declare them in the header doesn't matter).
+Any additional parameters that were passed but not listed in the ``Description`` object
+will be contained in the ``params`` kwarg as a dictionary. The validation of required
+parameters, coercion to the correct data type, and setting default values is all
+handled automatically for you based on the parameter descriptions in the ``Description``
+object passed. Two special methods of the ``Description`` object can be used for
+additional behavior control: :py:func:`girder.api.describe.Description.modelParam` and
+:py:func:`girder.api.describe.Description.jsonParam`.
+
+The ``modelParam`` method is used to convert parameters passed in as IDs to the corresponding
```

Comment:
Is it worth giving an example with two different models as params, so `id` and some other param need to be used together?  Because `id` does some syntactic sugar/magic for you that e.g. `folderId` wouldn't.

Also, maybe to make the docstring a bit more clear?

    The ``modelParam`` method is used to convert parameters passed in as IDs to the model document corresponding to those IDs,

or even `hydrated model document`.

🐃